File: bankmanagement.c
#include <stdio.h>
#include <stdlib.h>

struct Account {
    int accNo;
    char name[50];
    float balance;
};

void createAccount() {
    FILE *fp = fopen("bank.dat", "ab");
    struct Account a;

    printf("\nEnter Account Number: ");
    scanf("%d", &a.accNo);

    printf("Enter Account Holder Name: ");
    scanf(" %[^\n]", a.name);

    printf("Enter Initial Balance: ");
    scanf("%f", &a.balance);

    fwrite(&a, sizeof(a), 1, fp);
    fclose(fp);

    printf("\nAccount Created Successfully!\n");
}

void displayAccounts() {
    FILE *fp = fopen("bank.dat", "rb");
    struct Account a;

    printf("\n===== ACCOUNT DETAILS =====\n");

    while (fread(&a, sizeof(a), 1, fp)) {
        printf("\nAccount No : %d", a.accNo);
        printf("\nName       : %s", a.name);
        printf("\nBalance    : %.2f\n", a.balance);
    }

    fclose(fp);
}

void depositMoney() {
    FILE *fp = fopen("bank.dat", "rb+");
    struct Account a;
    int acc;
    float amount;

    printf("Enter Account Number: ");
    scanf("%d", &acc);

    while (fread(&a, sizeof(a), 1, fp)) {
        if (a.accNo == acc) {
            printf("Enter Deposit Amount: ");
            scanf("%f", &amount);

            a.balance += amount;

            fseek(fp, -sizeof(a), SEEK_CUR);
            fwrite(&a, sizeof(a), 1, fp);

            printf("Deposit Successful!\n");
            fclose(fp);
            return;
        }
    }

    printf("Account Not Found!\n");
    fclose(fp);
}

void withdrawMoney() {
    FILE *fp = fopen("bank.dat", "rb+");
    struct Account a;
    int acc;
    float amount;

    printf("Enter Account Number: ");
    scanf("%d", &acc);

    while (fread(&a, sizeof(a), 1, fp)) {
        if (a.accNo == acc) {

            printf("Enter Withdraw Amount: ");
            scanf("%f", &amount);

            if (amount > a.balance) {
                printf("Insufficient Balance!\n");
            } else {
                a.balance -= amount;

                fseek(fp, -sizeof(a), SEEK_CUR);
                fwrite(&a, sizeof(a), 1, fp);

                printf("Withdrawal Successful!\n");
            }

            fclose(fp);
            return;
        }
    }

    printf("Account Not Found!\n");
    fclose(fp);
}

int main() {

    int choice;

    do {
        printf("\n====== BANK MANAGEMENT SYSTEM ======\n");
        printf("1. Create Account\n");
        printf("2. Display Accounts\n");
        printf("3. Deposit Money\n");
        printf("4. Withdraw Money\n");
        printf("5. Exit\n");

        printf("Enter Choice: ");
        scanf("%d", &choice);

        switch(choice) {
            case 1:
                createAccount();
                break;
            case 2:
                displayAccounts();
                break;
            case 3:
                depositMoney();
                break;
            case 4:
                withdrawMoney();
                break;
            case 5:
                printf("Thank You!\n");
                break;
            default:
                printf("Invalid Choice!\n");
        }

    } while(choice != 5);

    return 0;
}

File: BankManagement.java
import java.util.Scanner;

class BankAccount {
    int accountNumber;
    String accountHolder;
    double balance;

    void createAccount(Scanner sc) {
        System.out.print("Enter Account Number: ");
        accountNumber = sc.nextInt();
        sc.nextLine();

        System.out.print("Enter Account Holder Name: ");
        accountHolder = sc.nextLine();

        System.out.print("Enter Initial Balance: ");
        balance = sc.nextDouble();

        System.out.println("\nAccount Created Successfully!");
    }

    void deposit(Scanner sc) {
        System.out.print("Enter Deposit Amount: ");
        double amount = sc.nextDouble();

        if (amount > 0) {
            balance += amount;
            System.out.println("Deposit Successful!");
        } else {
            System.out.println("Invalid Amount!");
        }
    }

    void withdraw(Scanner sc) {
        System.out.print("Enter Withdrawal Amount: ");
        double amount = sc.nextDouble();

        if (amount > balance) {
            System.out.println("Insufficient Balance!");
        } else if (amount <= 0) {
            System.out.println("Invalid Amount!");
        } else {
            balance -= amount;
            System.out.println("Withdrawal Successful!");
        }
    }

    void checkBalance() {
        System.out.println("Current Balance: ₹" + balance);
    }
}

public class BankManagement {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        BankAccount account = new BankAccount();

        int choice;

        do {
            System.out.println("\n===== BANK MANAGEMENT SYSTEM =====");
            System.out.println("1. Create Account");
            System.out.println("2. Deposit");
            System.out.println("3. Withdraw");
            System.out.println("4. Check Balance");
            System.out.println("5. Exit");
            System.out.print("Enter Choice: ");

            choice = sc.nextInt();

            switch (choice) {
                case 1:
                    account.createAccount(sc);
                    break;

                case 2:
                    account.deposit(sc);
                    break;

                case 3:
                    account.withdraw(sc);
                    break;

                case 4:
                    account.checkBalance();
                    break;

                case 5:
                    System.out.println("Thank You!");
                    break;

                default:
                    System.out.println("Invalid Choice!");
            }

        } while (choice != 5);

        sc.close();
    }
}